### PR TITLE
Add a parameter MaxFoodIntake

### PR DESCRIPTION
**Pull request by mdubied**
# Summary

This parameter is contained in the HealthInfo struct and limits the amount of food that an agent can physically eat. Choosing MaxFoodIntake large enough (4*tau=60 for example), we let the possibility for the agent to eat way more than needed, but still insure that they cannot eat the total amount of food on the platform (if curPlatFood>MaxFoodIntake)

### CODE REVIEW

**Comment by mdubied** [comment]
It seems that there is a problem on the frontend side. @udai-arneja can you have a closer look?

**Comment by udai-arneja** [comment]
I wrote some of the code so won't give a review, will wait for Jason

**Review by hussain2603 — DISMISSED** [review]

**Review by hussain2603 — COMMENTED** [review]
someone else should probs review frontend since I don't know that bit

**Review by jzzheng22 — COMMENTED** [review]
Code looks fine.

Shouldn't we make `MaxFoodIntake` dependent on `tau` though?

**Comment by mdubied** [comment]
We can, currently we can freely choose it on frontend. We chose the default paramters as 4*tau, where tau takes the best tuning we have so far

**Comment by jzzheng22** [comment]
> We can, currently we can freely choose it on frontend. We chose the default paramters as 4*tau, where tau takes the best tuning we have so far

I feel like it's better to just set MaxFoodIntake to always be 4*tau is all - what happens if someone changes tau but then doesn't change the maxFoodIntake?

**Comment by mdubied** [comment]
Yes, I am also happy with this solution

**Comment by udai-arneja** [comment]
We cant change tau from the frontend, I dont think we want to change it at all.

File: frontend/src/Components/NewRunConfig/Settings/Settings.tsx [hunk 1]
```diff
@@ -21,6 +21,7 @@ export default function Settings(props: SettingsInterface) {
     var key: Key = keyString; // converting keyString to type Key
     config[key] = value;
     setConfig(config);
+    console.log(config);
```
Comment (JaafarRammal) [inline frontend/src/Components/NewRunConfig/Settings/Settings.tsx:24]: I think a print that should be removed!

File: pkg/infra/baseAgent.go [hunk 2]
```diff
@@ -210,7 +210,7 @@ func (a *Base) TakeFood(amountOfFood food.FoodType) (food.FoodType, error) {
 	if amountOfFood < 0 {
 		return 0, &NegFoodError{}
 	}
-	foodTaken := food.FoodType(math.Min(float64(a.tower.currPlatFood), float64(amountOfFood)))
+	foodTaken := food.FoodType(utilFunctions.MinInt(int(a.tower.currPlatFood), int(amountOfFood), int(a.tower.healthInfo.MaxFoodIntake)))
```
Comment (JaafarRammal) [inline pkg/infra/baseAgent.go:213]: It means that agents are not allowed to destroy the food anymore (which is what selfish and random agents were doing).

**Review by JaafarRammal — DISMISSED** [review]
Code LGTM, left 2 minor comments

**Comment by jzzheng22** [comment]
> We cant change tau from the frontend, I dont think we want to change it at all. So you can change maxFoodIntake, but not tau. @jzzheng22

If the max food intake is based on tau, and you can't change tau, then surely max food intake shouldn't be changable?

**Comment by udai-arneja** [comment]
I think, while maxFood intake should typically be 4*tau, it would be interesting to see behaviour with the possibility of 'destroying food' vs not being able to destroy it... open to opinions @jzzheng22 @mdubied @JaafarRammal

File: pkg/infra/baseAgent.go (hunk 2 above)
Comment (udai-arneja) [reply pkg/infra/baseAgent.go:213]: Yep, unless we let them destroy food by setting the food intake to 100 or whatever the max is.... think this is a good point for experimentation... Open to opinions tho

File: frontend/src/Components/NewRunConfig/Settings/Settings.tsx (hunk 1 above)
Comment (udai-arneja) [reply frontend/src/Components/NewRunConfig/Settings/Settings.tsx:24]: Indeed my bad haahah

**Comment by mdubied** [comment]
I think we can leave it as the PR proposes, i.e. as a parameter on the frontend. We will probably run most of our simulations with the default value 4 tau, but as Udai mentioned, it can also be interesting to run simulations where maxFoodIntake="maxFoodOnPlatform"

**Comment by jzzheng22** [comment]
Happy with that then, fair enough.

**Review by jzzheng22 — APPROVED** [review]

File: frontend/src/Components/NewRunConfig/Settings/Settings.tsx (hunk 1 above)
Comment (mdubied) [reply frontend/src/Components/NewRunConfig/Settings/Settings.tsx]: print removed

**Review by JaafarRammal — APPROVED** [review]

**Pull request merged**